docs/user-guide/spark-user-guide.md

```diff
@@ -0,0 +1,159 @@
+
+# CQL Evaluation using Apache Spark
+
+## Overview
+
+The CQL evaluation function provided by the quality-measure-and-cohort-service project can be deployed in a variety of modes depending on the needs of the consuming solution. The most flexible option is a client-server mode of operation where the client performs the CQL evaluation for data that is stored in any remote FHIR server supporting the FHIR REST specification. While flexible, the client-server mode is likely not ideal for applications that need to process very large datasets due to the amount of data serialization and transfers required. For solutions wanting to process big data there is program provided that runs under Apache Spark. In this mode, the FHIR data model is eschewed in favor of traditional tabular data used by many analytics applications. Facilities are provided for loading the data from randomly ordered files, aggregated based on a specific execution context (e.g. Patient, Claim, etc.), and then evaluated using any number of CQL inputs. Outputs are written back out as tabular data. This guide describes how to configure and use that application.
```

Comment:
Would it be good to include a variation of the diagram we have showing various ways to run our CQL evaluation engine? One showing both the FHIR use case and Apache Spark use case, but maybe not the IBM FHIR embedded option?

Reply:
In my earliest draft of this paragraph I had assumed that we would put that diagram here, but I decided against it later because I think it focuses too much on things that aren't relevant to this guide specifically. I think we should have another page in the user guide that talks about the solution as a whole, includes that diagram, and links to this page for the specifics on Spark usage. I'd like to keep this page focused on the Spark implementation.